exercises/src/test/scala/fpinscala/parallelism/ParSpec.scala

```diff
@@ -0,0 +1,411 @@
+package fpinscala.parallelism
+
+import java.util.concurrent._
+import scala.concurrent.duration._
+import fpinscala.parallelism._
+
+import org.specs2.mutable.Specification
+import org.specs2.matcher.Matchers
+import org.specs2.specification.Scope
+import org.specs2.matcher.{Matchers,TerminationMatchers}
+
+class ParSpecification extends Specification with Matchers with TerminationMatchers {
+
+  trait ThreadPoolContext extends Scope {
+    private val startTime = System.currentTimeMillis
+    private val asyncThreadCount = new atomic.AtomicInteger
+    def elapsedTime = System.currentTimeMillis millis
+    val pool: ExecutorService = Executors.newCachedThreadPool(new ThreadFactory {
+      override def newThread(r: Runnable) = {
+        asyncThreadCount.incrementAndGet
+        Executors.defaultThreadFactory.newThread(r)
+      }
+    })
+    def threadCount = asyncThreadCount.get
+  }
+
+  class StringParOps(val s: String) {
```

Comment:
I had to remove `extends Anyval` to get this to work:
```diff
-  class StringParOps(val s: String) {
+  implicit class StringParOps(val s: String) {
```
This is the error reported by the compiler:
```
[error] /opt/fpinscala/exercises/src/test/scala/fpinscala/parallelism/ParSpec.scala:27: value class may not be a member of another class
[error]   implicit class StringParOps(val s: String) extends AnyVal {
[error]                  ^
[error] one error found
[error] (exercises/test:compileIncremental) Compilation failed
```